src/main/scala/org/constellation/storage/CheckpointService.scala

```diff
@@ -202,12 +265,15 @@ class CheckpointService[F[_]: Sync: LiftIO](
       }
     }
 
-  private def syncPending(cb: CheckpointBlock)(implicit dao: DAO): F[Unit] =
+  private def syncPending(storage: StorageService[IO, CheckpointBlock],
+                          cb: CheckpointBlock)(implicit dao: DAO): F[Unit] =
     Sync[F].delay {
-      pendingAcceptance.synchronized {
-        pendingAcceptance.contains(cb.baseHash).flatMap {
-          case false => pendingAcceptance.put(cb.baseHash, cb)
-          case _    => throw PendingAcceptance(cb)
+      storage.synchronized {
```

Comment:
Is it okay to block whole `storage` class?

Reply:
Only instance is being blocked.

Reply:
There problem is that we have one instance only for particular type of storage ;)

Reply:
and it will block other parts of the system as well

Reply:
it's single instance of StorageService not related to other services it needs to be blocked to handle concurrent updates, anyway I have migrated it to `Ref[F, Set[String]]`